### ISSUE TITLE
New license request: UCL-1.0 [SPDX-Online-Tools]

### ISSUE THREAD

**Issue by jlovejoy** [spdx/license-list-XML#894]
**1.** License Name: Upstream Compatibility License v1.0
**2.** Short identifier: UCL-1.0
**3.** License Author or steward: Larry Rosen & Nigel Tzeng
**4.** Comments: This license was approved by OSI in 2017 and somehow we missed it.
**5.** Standard License Header: Licensed under the Upstream Compatibility License 1.0
**6.** URL: https://opensource.org/licenses/UCL-1.0
**7.** OSI Status: Approved

**Comment by jlovejoy** [comment]
+1 to add - it's already OSI approved, we should have added it when it was approved by OSI, but missed it.

**Comment by swinslow** [comment]
This looks quite similar to the OSL-3.0 license -- reviewing to see whether the differences are substantive for SPDX matching guidelines purposes...

**Comment by jlovejoy** [comment]
difference is only in 1(c) - appears this was discussed on OSI mailing list (from someone on call looking back at those logs)

**Comment by swinslow** [comment]
Thanks @jlovejoy -- I'm +1 to add also, given that it's OSI approved and I agree the differences are sufficiently substantive to call it a different license from OSL-3.0 for SPDX purposes.

**Comment by pombredanne** [comment]
FWIW, this has been tracked in ScanCode for a while, but in hindsight there is no use in the wild I could find except for that one small project https://github.com/ftc-team-11553/ftc_2017 

All other references are to license tools (and several are to clones of scancode).
So this does not seem to be used much anywhere. @jlovejoy do you have some pointers to actual usages by chance?

Note that in any case since this is an officially OSI approved license, adding it even if not much used probably makes sense.

**Comment by swinslow** [comment]
@pombredanne I'm not seeing any usage in the wild other than that. But agreed, I think we've generally viewed that all OSI-approved licenses are appropriate to include on the list regardless of usage in the wild.

**Comment by swinslow** [comment]
Discussed on 2019-08-08 legal team call, agreed that this should be added.

**Comment by johnmhoran** [comment]
I'd be happy to work on this.

**Comment by swinslow** [comment]
Fantastic, thanks @johnmhoran!

**Comment by johnmhoran** [comment]
Hi @swinslow .  I'm belatedly tackling this issue, have just completed my initial draft of `UCL-1.0.xml`, and have one item to draw your attention to:

In subsection (ii) of Section 16 (Modification of This License), the OSL-3.0 license on which this is based, [as displayed by OSI](https://opensource.org/licenses/OSL-3.0), reads as follows:

```
(ii) You must replace the notice specified in the first paragraph above with the notice "Licensed under <insert your license name here>" . . . .
```

The [OSI version of UCL-1.0](https://opensource.org/licenses/UCL-1.0), on the other hand, omits the angle bracketed-language:

```
(ii) You must replace the notice specified in the first paragraph above with the notice "Licensed under " . . . .
```

I suspect this is inadvertent and have included the missing bracketed language in the new .xml file.  

What is our normal protocol in this sort of circumstance?  (I assume we need to contact the License Author/steward but meanwhile will push and open a PR once I've proofed etc.)

**Comment by swinslow** [comment]
Thanks @johnmhoran! Great catch on the missing bracketed language. I suspect you're correct and that the angle-bracketed language got dropped somewhere along the way -- perhaps getting mistaken for an HTML tag somewhere.

Agree for the moment with creating the PR _including_ the bracketed language you expect to see -- but I think this is something we'd want to raise with OSI for them to confirm / potentially correct on their site.

@jlovejoy, do you know who at OSI we should mention this to? Can discuss briefly at the beginning of the legal team call on Thursday to give context here...

**Comment by johnmhoran** [comment]
Thanks @swinslow.  Pushed and opened a PR but it looks like there are 2 build jobs (Java and Node.js) and the Java build failed -- see the [Travis error message](https://travis-ci.org/spdx/license-list-XML/jobs/580505299).

I'm not certain but this may be a user error.  I think the build failed because there was no test file, and I now seem to recall that I need to prepare one (and did not).  I'll do that next.

**Comment by johnmhoran** [comment]
@swinslow  I've pushed a `.txt` test file and it appears the Travis build was successful.

**Comment by swinslow** [comment]
@johnmhoran Thank you! Take a look at the commit I added in #922 and my comment in that thread. If that looks good to you, let me know and I'll go ahead and merge it. Thanks for your help with putting this one together!

**Comment by swinslow** [comment]
Merged in #922; closing issue
